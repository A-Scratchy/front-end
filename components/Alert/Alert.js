import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import ScreenReaderOnly from 'components/_common_/ScreenReaderOnly/ScreenReaderOnly';
import styles from './Alert.css';

Alert.propTypes = {
  children: PropTypes.node.isRequired,
  isOpen: PropTypes.bool,
  onClose: PropTypes.func,
  type: PropTypes.oneOf(['error', 'success', 'warning']),
};

Alert.defaultProps = {
  isOpen: true,
  onClose: undefined,
  type: 'error',
};

function Alert({ children, isOpen, onClose, type }) {
  return (
    <div
      className={classNames(styles.Alert, {
        [styles.error]: type === 'error',
        [styles.success]: type === 'success',
        [styles.warning]: type === 'warning',
        [styles.visible]: isOpen,
      })}
      role="alert"
    >
      {Boolean(onClose) && isOpen ? (
        <button type="button" className={styles.alertCloseButton} onClick={onClose}>
          <ScreenReaderOnly>Close Alert</ScreenReaderOnly>
          <span>&times;</span>
        </button>
      ) : null}
      {children}
    </div>
  );
}

export default Alert;
