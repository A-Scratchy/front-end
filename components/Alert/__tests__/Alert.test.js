/* eslint-env jest */
import React from 'react';
import { mount } from 'enzyme';
import createSnapshotTest from 'test-utils/createSnapshotTest';

import Alert from '../Alert';

describe('Alert', () => {
  it('should render with required props', () => {
    createSnapshotTest(<Alert>Error Test Alert!</Alert>);
  });

  it('should render with many props assigned', () => {
    createSnapshotTest(
      <Alert onClose={jest.fn} type="success" isOpen>
        Success Test Alert!
      </Alert>,
    );
  });

  it('should call props.onClose when isOpen and when CloseAlertButton is clicked', () => {
    const onCloseMock = jest.fn();

    const wrapper = mount(
      <Alert onClose={onCloseMock} type="success" isOpen>
        Success Test Alert!
      </Alert>,
    );

    const CloseAlertButton = wrapper.find('button');

    expect(onCloseMock).toHaveBeenCalledTimes(0);
    CloseAlertButton.simulate('click');
    expect(onCloseMock).toHaveBeenCalledTimes(1);
  });

  it('should not render button if isOpen is false or no onClose function passed', () => {
    const AlertNotOpen = mount(<Alert isOpen={false}>Testing!</Alert>);
    const AlertNoOnClose = mount(<Alert isOpen>Testing!</Alert>);

    expect(AlertNotOpen.find('button')).not.toExist();
    expect(AlertNoOnClose.find('button')).not.toExist();
  });
});
