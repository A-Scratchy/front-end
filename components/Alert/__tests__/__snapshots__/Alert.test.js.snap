// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should render with many props assigned 1`] = `
<aside
  className="Alert success visible"
  role="alert"
>
  <button
    className="alertCloseButton"
    onClick={[Function]}
    type="button"
  >
    <span
      className="ScreenReaderOnly"
    >
      Close Alert
    </span>
    <span>
      Ã—
    </span>
  </button>
  Success Test Alert!
</aside>
`;

exports[`Alert should render with required props 1`] = `
<aside
  className="Alert error visible"
  role="alert"
>
  Error Test Alert!
</aside>
`;
